### PR TITLE
feat: add Stash Staged option for Source Control view

### LINKED ISSUE #149149: Support `git stash --staged` in the Source Control view
A few months ago, Git introduced a new `--staged` option for the `stash` subcommand to stash away what has been added to the index (and nothing else).

The GitHub Blog post: https://github.blog/2022-01-24-highlights-from-git-2-35/

It would be nice if it is available in the Source Control view.

### PR DESCRIPTION
**Pull request by liby**
This PR fixes #149149

## More information
With Git 2.35 (Q1 2022), [`git stash`](https://github.com/git/git/blob/44ac8fd1b4ce7598e2e5f4045dfbb6593a3f5b84/Documentation/git-stash.txt)([man](https://git-scm.com/docs/git-stash)) learned the `--staged` option to stash away what has been added to the index (and nothing else).

<details>
<summary> Screenshot </summary>

Menu
Command

</details>

### CODE REVIEW

**Comment by liby** [comment]
@lszomoru TBR

File: extensions/git/src/main.ts [hunk 1]
```diff
@@ -245,6 +245,57 @@ async function checkGitv1(info: IGit): Promise<void> {
...
 }
 
+
+const cmpOrdering = {
+	'Less': -1,
+	'Equal': 0,
+	'Greater': 1,
+} as const;
+
+const compareVersion = (localVersion: string, specificVersion: string) => {
```
Comment (lszomoru) [inline extensions/git/src/main.ts:255]: There is already a method, `compareGitVersionTo()` to compare Git versions.

**Comment by AgainPsychoX** [comment]
I hope it merges soon 🙏

**Comment by liwenkang** [comment]
It' useful for me

**Comment by chalkygames123** [comment]
Hi, Is there any ETA?

**Comment by chalkygames123** [comment]
@lszomoru Friendly ping :)

**Comment by lszomoru** [comment]
@liby, sorry for not getting back to you on this. I completely forgot about this contribution and today I was doing some work around the stash commands so went ahead and implemented this functionality here - https://github.com/microsoft/vscode/pull/169539. This capability should be available in the next Insiders release.

**Pull request closed without merging**